File: Domain/Goals/Goal.cs
using Domain.Enums;

namespace Domain.Goals
{
    public abstract class Goal
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool CustomUnit { get; set; }
        public string Unit { get; set; }
        public DateOnly Deadline { get; set; }
        public GoalStatus Status { get; set; }
    }
}

File: goals-progress/src/interfaces/Goal.ts
import { ExtendedProgress, StandardProgress } from "./Progress"

export interface Goal {
    id: number,
    name: string,
    description: string,
    currentValue: number,
    targetValue: number,
    customUnit: boolean,
    unit: string,
    deadline: string,
    status: number,
    type: number
}

export interface StandardGoal extends Goal{
    progresses: StandardProgress[]
}

export interface ExtendedGoal extends Goal {
    progresses: ExtendedProgress[]
}